### PR TITLE
Add initial unit tests

### PR DESCRIPTION
**Pull request by networkfusion**
## Description
This library did not contain unit tests.
Adds Initial tests using big brother implementation from: https://github.com/dotnet/runtime/tree/main/src/libraries/System.IO.FileSystem.Primitives/tests

Further tests "could" be obtained from https://github.com/dotnet/runtime/tree/main/src/libraries/System.IO.FileSystem/tests

## Motivation and Context
* Project required some unit tests.
* Library should start to align with `.Net` as far as possible.

## How Has This Been Tested?
All changes should be non intrusive.

## Types of changes

- [x] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue with code or algorithm)
- [ ] New feature (non-breaking change which adds functionality to code)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [x] Unit Tests (add new Unit Test(s) or improved existing one(s), has no impact on code or features)
- [x] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist:

- [ ] My code follows the code style of this project (only if there are changes in source code).
- [ ] My changes require an update to the documentation (there are changes that require the docs website to be updated).
- [ ] I have updated the documentation accordingly (the changes require an update on the docs in this repo).
- [ ] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/master/CONTRIBUTING.md) document.
- [ ] I have tested everything locally and all new and existing tests passed (only if there are changes in source code).
- [ ] I have added new tests to cover my changes.

### CODE REVIEW

**Comment by networkfusion** [comment]
Note that FileAttributes are (actually) 2 bytes in length (fixed) and may possibily require a change in the native layer.

File: System.IO.FileSystem/tests/System.IO.FileSystem.Tests/Properties/AssemblyInfo.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
+﻿using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+// General Information about an assembly is controlled through the following 
+// set of attributes. Change these attribute values to modify the information
+// associated with an assembly.
+[assembly: AssemblyDescription("")]
```
Comment (Ellerbach) [inline System.IO.FileSystem/tests/System.IO.FileSystem.Tests/Properties/AssemblyInfo.cs:8]: you should adjust this file

File: System.IO.FileSystem/tests/System.IO.FileSystem.Tests/nano.runsettings [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    <MaxCpuCount>1</MaxCpuCount>
+    <ResultsDirectory>.\TestResults</ResultsDirectory>
+    <!-- Path relative to solution directory -->
+    <TestSessionTimeout>120000</TestSessionTimeout>
+    <!-- Milliseconds -->
+    <TargetFrameworkVersion>Framework40</TargetFrameworkVersion>
+  </RunConfiguration>
+  <nanoFrameworkAdapter>
+    <Logging>None</Logging>
+    <IsRealHardware>False</IsRealHardware>
```
Comment (Ellerbach) [inline System.IO.FileSystem/tests/System.IO.FileSystem.Tests/nano.runsettings:14]: are the tests running in the virtual device? Is the System.IO.File having a native implementation supported there?

**Review by Ellerbach — COMMENTED** [review]
Good start, few comments

File: System.IO.FileSystem/tests/System.IO.FileSystem.Tests/Properties/AssemblyInfo.cs (hunk 1 above)
Comment (networkfusion) [reply System.IO.FileSystem/tests/System.IO.FileSystem.Tests/Properties/AssemblyInfo.cs]: Done.

File: System.IO.FileSystem/tests/System.IO.FileSystem.Tests/nano.runsettings (hunk 2 above)
Comment (networkfusion) [reply System.IO.FileSystem/tests/System.IO.FileSystem.Tests/nano.runsettings]: Currently the tests are just checking that the enumerations are correct (in theory) aligned to the full fat .NET implementation. We need to make sure these can align with the native etc..

**Comment by networkfusion** [comment]
Failing (like other libs) due to the latest test framework release!

**Comment by josesimoes** [comment]
OK and kudos for adding unit tests! Always welcome.

OK to for making the enum type match the full .NET ones.
Now, I don't see a point on adding all these attributes that we can't support... 
If this is for code sharing use cases (with code files shared in full .NET projects) adding them will give the wrong impression that code is fine. 
On the contrary if we don't have them, build will break and the developer will have to handle those situations.

**Comment by josesimoes** [comment]
> I see your point, but would rather see a not supported exception for them. 

Can we throw an exception for a non existing enum item? 😯   I don't think I've seen that... 🤔
The only usable way I see is just not having them there.

> For the sake of this PR, I think a comprimise is to comment them out and add a note (which I have now done).
> We can always remove them on the next pass as more unit tests are added.
On a quick look: we won't ever have support for those enums. I guess that was the reason for not having them there in the 1st place.

**Comment by networkfusion** [comment]
> > I see your point, but would rather see a not supported exception for them.
> 
> Can we throw an exception for a non existing enum item? 😯 I don't think I've seen that... 🤔 The only usable way I see is just not having them there.
> 
> > For the sake of this PR, I think a comprimise is to comment them out and add a note (which I have now done).
> > We can always remove them on the next pass as more unit tests are added.
> > On a quick look: we won't ever have support for those enums. I guess that was the reason for not having them there in the 1st place.

Pesudo code:
```
        /// <summary>
        /// Sets the specified FileAttributes of the file on the specified path.
        /// </summary>
        /// <param name="path">The path to the file.</param>
        /// <param name="fileAttributes">A bitwise combination of the enumeration values.</param>
        public static void SetAttributes(string path, FileAttributes fileAttributes)
        {
            if ((byte)fileAttributes > (byte)FileAttributes.Archive)
            {
              throw new NotSupportedException();
            }
            else
            {
                SetAttributesNative(path, (byte)fileAttributes);
            }
        }
```

**Comment by josesimoes** [comment]
@networkfusion I get that you can add those checks on the library code. My concern is with the code calling our library. We can't enforce those to be checked. 
Simplest fix for this and not having to worry about it: don't have them in the 1st place.

Any user code using those enums that have no implementation it will build succesfully but those can not be honoured (whathever they are doing).

**Comment by networkfusion** [comment]
To enable progress, I have removed the concern and moved it to a seperate branch.
Although the PR is now failing for something else 😓

**Comment by networkfusion** [comment]
Closing this as #98 is adding them.

**Pull request closed without merging**